Release v0.0.7

alienc0der opened the pull request.

alienc0der commented:
@KingGorrin @Sol-Sanctum check the last commit.

KingGorrin commented:
Same result as in #22

Sol-Sanctum commented:
@KingGorrin 

> \screen_capturer_plugin.cpp(10,10): fatal  error C1083: Cannot open include file: 'atlimage.h': No such file or directory

This might be due to a missing dependency. I was able to build the project with IntelliJ IDEA.

Related: https://stackoverflow.com/a/46304843

KingGorrin commented:
That did the trick! Thanks @Sol-Sanctum

alienc0der commented:
> @KingGorrin
> 
> > \screen_capturer_plugin.cpp(10,10): fatal  error C1083: Cannot open include file: 'atlimage.h': No such file or directory
> 
> This might be due to a missing dependency. I was able to build the project with IntelliJ IDEA.
> 
> Related: https://stackoverflow.com/a/46304843

Indeed it requires [Windows C++ ATL](https://github.com/leanflutter/screen_capturer#windows-requirements)

Sol-Sanctum commented:
So far, most things are working as expected on Windows.

Tested
- WC with the Demo and Bridge sites*
- Wallet backup process

\* I'm able to consistently establish connections with the URL and screenshot methods, but the automatic Syrius Desktop trigger doesn't work often.

We may want to create an issue to keep track of this result and address it at a later time.

sumamu commented:
@alienc0der you need to request reviews before anyone can approve the PR.

Sol-Sanctum commented:
What's in this release?

KingGorrin commented:
@alienc0der can you provide a comprehensive PR description of what this change entails and why?

alienc0der commented:
- Updated dependencies
- Updated `WalletConnect` integration
- Fixed charts and duplicated nodes in `Node Management`
- Fixed `metadata.dart` when single quotes are present in the commit message
- Added possibility to turn off the auto-receiver
- Added possibility to open the `logs` directory
- Added hardcoded community nodes
- UX/UI improvements: badge for notifications icon

Sol-Sanctum commented:
Please add your list to the first post in this thread

Here are some thoughts about the PR:
* **UX/UI improvements: badge for notifications icon**
   * We should be able to acknowledge (read) notifications without deleting them.
   * An "Acknowledge all" option should be offered.
   
* Node Management badges are misaligned

* **Added possibility to turn off the auto-receiver**
   * Is this phrasing correct? 
   "Disable the auto-receiver to protect against dusting attacks" -- This implementation doesn't do that. 
   * The UI doesn't indicate any pending inbound transfers or allow users to cherry-pick transactions. It effectively prevents users from receiving anything.
   * When I re-enabled it, I saw N number of "error -32000: account-block from-block already received" notifications, one for each pending tx. Not sure how often that will happen.
   * After it's re-enabled, wallets might be generating plasma for numerous transactions back-to-back. Is this acceptable UX?
   
* **Added hardcoded community nodes**
   * Please add more than one: wss://node.zenonhub.io:35998
   * I believe these should not be hardcoded in the wallet. If it's not changed now, then we should create a backlog item/ticket to address this.
   
* Can we disable the pulsating dot at the top right?

* I like the lightning bolt and WalletConnect seems to work.

coinselor commented:
I've tried running the release from https://github.com/alienc0der/syrius/releases updated yesterday, which I assumed is the correct one this PR references. Unfortunately, I got the following error and the wallet window is just blank:

`[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Null check operator used on a null value
#0      Web3WalletService.getWeb3Wallet (package:zenon_syrius_wallet_flutter/services/web3wallet_service.dart:124)
#1      new NoMService (package:zenon_syrius_wallet_flutter/blocs/wallet_connect/chains/nom_service.dart:58)
#2      setup (package:zenon_syrius_wallet_flutter/main.dart:177)
#3      main (package:zenon_syrius_wallet_flutter/main.dart:79)
<asynchronous suspension>
`

OS: Ubuntu 23.04 x86_64 

Any extra info/help testing a new release, let me know. Happy to help.

digitalSloth commented:
I tried the same release as coinselor but on OS X and got a similar issue, syrius opens to a black screen and freezes, I have to force quit it.

The following is from `syrius/log` 

```
INFO WalletConnectService kWcProjectId missing 2023-09-14 21:10:47.922718: null null
```

KingGorrin commented:
I've tried running the Windows release from [releases](https://github.com/alienc0der/syrius/releases/tag/v0.0.7-alphanet) updated 4 days ago and got a blank screen when starting it up. Tried various things, even deleting the whole `znn` folder does not help.

The log only show the following line:

`INFO WalletConnectService kWcProjectId missing 2023-09-16 18:14:08.116974: null null`

I was able to build it from source and run it., but fails when trying to wrap 100 QSR. "Unable to make request and couldn't retry." The log has an `-32000: account-block signature is invalid` error.
